### DISCUSSION TITLE
Unable to get QGIS environment working in own infrastructure.

### DISCUSSION THREAD

**Discussion by stviaene-si** [microsoft/PlanetaryComputer#271] [Q&A]
Hi all,

We've set up our own copy of the Planetary Computer Hub based on the code in the [planetary-computer-hub repo](https://github.com/microsoft/planetary-computer-hub) (brilliant, thanks for making that public!), having built our own images off the definitions provided in the [planetary-computer-containers repo](https://github.com/microsoft/planetary-computer-containers).

This all works perfectly, apart from the QGIS desktop. Whenever we request this environment, we just get a jupyter portal with a file navigator. Navigating to the `/desktop` gives an empty screen with the message "Something went wrong, connection is closed" at the top. I've tried running the container locally as well, and I get the same result. The container log and VNC log are attached, I've not managed to make much sense of them.

Any ideas what might be going on here?

Thanks!
[2023-09-27Vnc.log](https://github.com/microsoft/PlanetaryComputer/files/12736532/2023-09-27Vnc.log)
[2023-09-27Container.log](https://github.com/microsoft/PlanetaryComputer/files/12736533/2023-09-27Container.log)

**Comment by TomAugspurger** [comment]
Mmm I don't have a great idea. https://discourse.pangeo.io/t/run-linux-desktop-apps-in-mybinder-org-your-jupyterhub/1978 might have some links you can follow.
